dynamics/client.py

```diff
@@ -173,17 +184,23 @@ def from_environment(cls):
         * DYNAMICS_CLIENT_ID: client id string
         * DYNAMICS_CLIENT_SECRET: client secret key string
         * DYNAMICS_SCOPE: comma separated list of urls
+        * DYNAMICS_RESOURCE: single target url
+
+        Only specify SCOPE or RESOURCE, but not both.
 
         :raises KeyError: An environment variable was not configured properly
         """
 
-        api_url = os.environ["DYNAMICS_API_URL"]
-        token_url = os.environ["DYNAMICS_TOKEN_URL"]
-        client_id = os.environ["DYNAMICS_CLIENT_ID"]
-        client_secret = os.environ["DYNAMICS_CLIENT_SECRET"]
-        scope = os.environ["DYNAMICS_SCOPE"].split(",")
+        api_url = os.environ.get("DYNAMICS_API_URL")
```

Comment:
I would not use `.get(...)` for `api_url`, `token_url`, `client_id`, and `client_secret`, since all of those are mandatory, as indicated by the `__init__` typing.

Reply:
Although the __init__ for the DynamicsClient handles the new resource/scope init logic, I complied with the `KeyError` structure here to ensure appropriate environemnt variables are set and a helpful message comes back.

Reply:
[This here](https://github.com/MrThearMan/dynamics-client/pull/4/commits/c6471ba969439c92f488ff2a5454c8c2bb7c4549#diff-fafb0b7303b9394168b2c6f6aef7be16bcc89b26745d80f08eaa4f03eb37bb5bR211) is great, but the latest commit changed it so that both are mandatory, see my [comment here.](https://github.com/MrThearMan/dynamics-client/pull/4/commits/12112b6655ccdbedfa8650d7a8239a7226c2c949#r1148505014)